syntax = "proto3";

package transaction_verification;

service TransactionVerificationService {
    rpc VerifyTransaction (TransactionVerificationRequest) returns (TransactionVerificationResponse);
}

message TransactionVerificationRequest {
    User user = 1;
    CreditCard credit_card = 2;
    string user_comment = 3;
    Address billing_address = 4;
}

message TransactionVerificationResponse {
    bool is_verified = 1;
}

message User {
    string name = 1;
    string email = 2;
}

message CreditCard {
    string number = 1;
    string expiration_date = 2;
    string cvv = 3;
}

message Address {
    string street = 1;
    string city = 2;
    string state = 3;
    string zip = 4;
    string country = 5;
}
